### PR TITLE
Upgrade to sbt 1.1.1

### PR DESCRIPTION
**Pull request by by-dam**

### CODE REVIEW

**Comment by carlpulley** [comment]
Is it sensible to think about doing this yet?

My issue is that SBT 1.X is very new and tons of issues have been uncovered with this (some have been fixed, but other serious issues will lurk and cause lost time on projects debugging them). So I'd argue that we should wait longer before this ticket is merged.

If there is a pressing need for this work, then this code should be cross-SBT compiled so that other teams may use it.

File: src/main/scala/net/cakesolutions/CakeBuildInfoPlugin.scala
Comment (carlpulley) [inline src/main/scala/net/cakesolutions/CakeBuildInfoPlugin.scala]: Why are we removing the Scaladoc?

File: src/main/scala/net/cakesolutions/CakeBuildPlugin.scala
Comment (carlpulley) [inline src/main/scala/net/cakesolutions/CakeBuildPlugin.scala]: Surely `Vector()` or better `Vector.empty`?

File: src/main/scala/net/cakesolutions/CakePublishMavenPlugin.scala
Comment (carlpulley) [inline src/main/scala/net/cakesolutions/CakePublishMavenPlugin.scala]: Can't we factor these SBT version numbers away from here? Changing the SBT version value shouldn't have a surprising change here.

**Comment by simonsouter** [comment]
Might be worth keeping this on a separate 1.x branch of sbt-cake and releasing it separately?  Until we decide to migrate over en masse.

File: src/sbt-test/sbt-cake/testrunner/build.sbt
Comment (carlpulley) [inline src/sbt-test/sbt-cake/testrunner/build.sbt]: Shouldn't this value be updated?

File: src/main/scala/net/cakesolutions/CakePublishMavenPlugin.scala
Comment (by-dam) [reply src/main/scala/net/cakesolutions/CakePublishMavenPlugin.scala]: Do you mean getting the credentials directly in `~/.sbt` ?
`(Path.userHome / ".sbt" / ".credentials")`

File: src/sbt-test/sbt-cake/testrunner/build.sbt
Comment (by-dam) [reply src/sbt-test/sbt-cake/testrunner/build.sbt]: Akka version didn't change in this PR so I guess not

File: project/project/CakePlatformDependencies.scala
Comment (emilypi) [inline project/project/CakePlatformDependencies.scala]: `4.1.21.Final` is out, and stable.

File: project/project/CakePlatformDependencies.scala
Comment (emilypi) [inline project/project/CakePlatformDependencies.scala:115]: We should plan on incorporating `scalafix` as well

File: src/sbt-test/sbt-cake/release-notes/build.sbt
Comment (emilypi) [inline src/sbt-test/sbt-cake/release-notes/build.sbt]: why `s`?

**Comment by emilypi** [comment]
I can make another pass at this over the weekend. There are a few updates I'd like to request.

File: project/project/CakePlatformDependencies.scala
Comment (by-dam) [reply project/project/CakePlatformDependencies.scala:115]: I agree but would do in another PR

**Review by emilypi — APPROVED** [review]

**Comment by emilypi** [comment]
@fommil if we get another set of approvals, I think we're good to merge this one.

**Comment by fommil** [comment]
How many approvals does it need? I have merge rights and probably publish rights, but that might be automated. Say the word.

**Comment by emilypi** [comment]
If you're good with it, we can merge. Usually it's 2 on the most recent changes, not 2 in general, but I think for this, we're good to merge. The sooner the better.

**Comment by fommil** [comment]
I'll give @carlpulley a day to object. If there is silence, I'll merge. If somebody else merges, then my hands are clean :stuck_out_tongue:

**Comment by emilypi** [comment]
@carlpulley @fommil @simonsouter @by-dam Can we merge this, or are we waiting on something in particular?

**Comment by by-dam** [comment]
I'm ok to merge but @carlpulley had concerns about sbt 1.x maturity.

**Pull request merged**
